### PR TITLE
(feat): Add head management

### LINKED ISSUE #74: Setup head management + display note title
- setup `<head>` management application service. It will be used to set title, description and meta-tags (open graph data, etc)
- consider using https://github.com/vueuse/head tool 
- once head management is set up, use it to display opened note title (first text block trimmed by 50 chars) in browser window title

### PR DESCRIPTION
**Pull request by HyTekCoop**
Resolves #74 

Change of site titles in the browser has been added. Titles have been changed for Homepage, individually for each note and for new note, note settings and user settings.
And also function for limit title to 50 chars have been added.

### CODE REVIEW

File: src/presentation/pages/Note.vue [hunk 1]
```diff
@@ -46,6 +52,23 @@ function noteChanged(data: NoteContent): void {
...
+
+/**
+ * Changing the title in the browser
+ */
+if (!props.id) {
+  useHead({
+    title: t('site.titles.newNote'),
+  });
+} else {
+  watchEffect(() => {
```
Comment (neSpecc) [inline src/presentation/pages/Note.vue:64]: useHead is not reactive?

File: src/application/services/useBrowserTitle.ts
Comment (neSpecc) [inline src/application/services/useBrowserTitle.ts:1]: seems redundant

File: src/application/services/useNote.ts [hunk 2]
```diff
@@ -87,6 +92,13 @@ export default function (options: UseNoteComposableOptions): UseNoteComposableSt
    */
   const router = useRouter();
 
+  /**
+   * Title identifier
+   */
+  const title = computed(() => {
+    return note.value?.content.blocks[0].data.text;
```
[the remark this answers is not in the document]
Comment (HyTekCoop) [reply src/application/services/useNote.ts]: added check for the existence of the block and that it contains text

File: src/application/services/useBrowserTitle.ts
Comment (HyTekCoop) [reply src/application/services/useBrowserTitle.ts:1]: function moved to useNote

File: src/presentation/pages/Note.vue (hunk 1 above)
Comment (HyTekCoop) [reply src/presentation/pages/Note.vue:64]: At the moment we decided to leave this function watchEffect

File: src/application/services/useNote.ts [hunk 3]
```diff
@@ -92,6 +97,20 @@ export default function (options: UseNoteComposableOptions): UseNoteComposableSt
    */
   const router = useRouter();
 
+  /**
+   * Title identifier
+   */
+  const limitChars = 50;
```
Comment (neSpecc) [inline src/application/services/useNote.ts:103]: variable naming is unclear for its scope (`useNote`)

File: src/application/services/useNote.ts [hunk 4]
```diff
@@ -92,6 +97,20 @@ export default function (options: UseNoteComposableOptions): UseNoteComposableSt
...
+  const title = computed(() => {
+    const firstBlockNote = note.value?.content.blocks[0];
```
Comment (neSpecc) [inline src/application/services/useNote.ts:105]: naming is unclear as well

**Review by neSpecc — CHANGES_REQUESTED** [review]

File: src/application/services/useNote.ts (hunk 3 above)
Comment (HyTekCoop) [reply src/application/services/useNote.ts]: Rename variables

File: src/application/services/useNote.ts (hunk 4 above)
Comment (HyTekCoop) [reply src/application/services/useNote.ts]: Rename variables

**Review by neSpecc — APPROVED** [review]

**Review by slaveeks — APPROVED** [review]

**Pull request merged**